apps/angular-store-app/src/app/cart/state/cart.effects.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@angular/core';
+import { Actions, Effect, ofType } from '@datorama/akita-ng-effects';
+import { CartActions } from './cart.actions';
+import { map, switchMap, tap } from 'rxjs/operators';
+import { CartStore } from './cart.store';
+import { of, timer } from 'rxjs';
+
+@Injectable()
+export class CartEffects {
+  constructor(private actions$: Actions, private cartStore: CartStore) {
+    console.log(this);
+  }
+
+  @Effect()
+  removeItem$ = this.actions$.pipe(
+    ofType(CartActions.removeItem),
+    tap((val) => console.log('[Cart Effects] removeItem$', val)),
+    switchMap((test) => {
+      // do async operations
+      this.cartStore.remove(test.productId);
+      return of(this.actions$.dispatch(CartActions.removeItemSuccess()));
```

Comment:
I think it'll be better to follow the `ngrx/effects` style that does the following:
- By default, it expects the `effect` to return a new action and dispatches it internally
- You can pass `{ dispatch: false }` as the second argument to cancel the previous behavior.

Reply:
Good call.
When this pull request is merged I might have a few more feature adaptions for the effects.
We can chat later about this.